### DISCUSSION TITLE
Latest MRTK3 drop (pre-17) dependencies clash with Latest Azure Spatial Anchors ( 2.14.1)

### DISCUSSION THREAD

**Discussion by LocalJoost** [microsoft/MixedRealityToolkit-Unity#11673] [General]
## Describe the bug

When you install MRTK3 pre-17 in a clean project and configure it, then add Azure Spatial Anchors, you will get 143 errors of the type:
**GUID [300c4804f47d540fc9efec07999accf7] for asset 'Packages/com.unity.xr.arsubsystems/Editor' conflicts with:
  'Packages/com.unity.xr.arfoundation/Editor/ARSubsystems' (current owner)
We can't assign a new GUID because the asset is in an immutable folder. The asset will be ignored.**

## To reproduce
* Create a new project, then close Unity
* Using the MRTK feature tool, install all MRTK3 components, the experimental MRTK3 Data Binding and Theming pre-17 component, OpenXR 1.7.0, and Spatial Audio 2.0.37 
* Re-open new project with Unity, accept the question about the new Input System, and updating to new XR InteractionLayerMask. Editor restarts both times.
* Project has 0 errors.
* Switch to UWP build platform
*  Project has 0 errors.
* Close Unity again
* Using the MRKT feature tool Add Azure Spatial Anchors SDK core 2.14.1 and Azure Spatial Anchors SDK for Windows 2.14.1
* Open project in Unity
* Project has 143 errors

## Expected behavior

No errors ;)

## Screenshots
![image](https://github.com/microsoft/MixedRealityToolkit-Unity/assets/4129183/0991300b-1ffe-4565-b3b1-0c411eaf3d66)

## Setup
- Unity Version 2021.3.23f1
- MRTK Version 3.0.0 pre-17, including MRTK3 Data Binding and Theming

## Target platform
- HoloLens 2

## Additional context

Repo : https://github.com/LocalJoost/TestConflict3.git

**Comment by shaynie** [comment]
@LocalJoost  - Thanks for the bug.  This is caused by the Azure Spatial Anchors core package pulling in com.unity.xr.arsubsystems v4.0.8 as a dependency.  The arsubsystems package was deprecated and its functionality was moved in to the ARFoundation package in version 5.0.0.  Since the MRTK3 input package depends on ARFoundation 5.0.5, you see the conflicting class GUIDs if a pre 5.x ARSubsystems package is imported.  The workaround is to manually upgrade the ARSubsystems package in your project to 5.0.2,  which just removes all the conflicting classes.  

Workaround:  In Unity, open the Package Manager window, and select the drop down arrow next to the '+'.  Select 'Add package by name", and then enter 'com.unity.xr.arsubsystems' in the name field, and '5.0.2' in the version field.

I'm converting this issue to a discussion so that others can find the workaround.  I'm also requesting that the ASA team upgrade their ARFoundation requirement to 5.x+ to avoid this in their next release.

**Reply by LocalJoost** [reply]
Awesome. Thanks for the workaround!

**Comment by AMollis** [comment]
Keeping discussion opened until ASA issue is resolved.
